packages/@aws-cdk/aws-ecs/lib/ec2/ec2-service.ts

```diff
@@ -67,19 +67,15 @@ export interface Ec2ServiceProps extends BaseServiceProps {
   readonly daemon?: boolean;
 }
 
-export interface IEc2Service extends IService {
-
-}
-
 /**
  * Start a service on an ECS cluster
  *
  * @resource AWS::ECS::Service
  */
-export class Ec2Service extends BaseService implements IEc2Service, elb.ILoadBalancerTarget {
+export class Ec2Service extends BaseService implements IService, elb.ILoadBalancerTarget {
```

Comment:
This is actually intentional, and supposed to be enforced by awslint. The rational is that otherwise it will be a breaking change in case we want to extend the base interface in one of the derived ones.

Reply:
Hmm, I told them to do it. I don't believe it will be a breaking change to add an intermediary interface later on if we ever wanted to do that (which we probably won't).

In what situation would be it be breaking? As far as I can tell, uses of this interface are currently in output position, meaning we can narrow the type without breaking the API.